### PR TITLE
Fix PMD/Build issues

### PR DESCRIPTION
**Pull request by matiasroldan6**
Fixes #

## Specific Changes

  - 
  -
  -

### CODE REVIEW

File: libraries/bot-builder/src/main/java/com/microsoft/bot/builder/ChannelServiceHandler.java [hunk 1]
```diff
@@ -58,6 +57,8 @@ public ChannelServiceHandler(
      * HttpClient)} , we should move this code somewhere in that library when
      * we refactor auth, for now we keep it private to avoid adding more public
      * static functions that we will need to deprecate later.
+     * @param authHeader
+     * @return createAnonymousSkillClaim().
```
Comment (Batta32) [inline libraries/bot-builder/src/main/java/com/microsoft/bot/builder/ChannelServiceHandler.java:61]: Is it possible to add a comment to these?

File: libraries/bot-builder/src/main/java/com/microsoft/bot/builder/CloudAdapterBase.java [hunk 2]
```diff
@@ -199,7 +199,10 @@ protected CloudAdapterBase(BotFrameworkAuthentication withBotFrameworkAuthentica
             return Async.completeExceptionally(new IllegalArgumentException("reference"));
         }
 
-        return processProactive(createClaimsIdentity(botAppId, null), reference.getContinuationActivity(), null, callback);
+        return processProactive(createClaimsIdentity(botAppId, null),
+                                reference.getContinuationActivity(),
+                        null,
```
Comment (Batta32) [inline libraries/bot-builder/src/main/java/com/microsoft/bot/builder/CloudAdapterBase.java:204]: Can we check the indentation?

File: libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java [hunk 3]
```diff
@@ -68,13 +70,28 @@ public SkillHandlerImpl(
         this.getOAuthScope = withGetOAuthScope;
     }
 
+    /**
+     *
+     * @param claimsIdentity
+     * @param conversationId
+     * @param activity
+     * @return processActivity().
+     */
```
Comment (Batta32) [inline libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java:79]: Can we add comments?

File: libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java [hunk 4]
```diff
@@ -68,13 +70,28 @@ public SkillHandlerImpl(
...
     }
 
+    /**
+     *
+     * @param claimsIdentity
+     * @param conversationId
+     * @param activityId
+     * @param activity
+     * @return processActivity().
+     */
```
Comment (Batta32) [inline libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java:94]: Can we add comments?

File: libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java [hunk 5]
```diff
@@ -83,6 +100,13 @@ public SkillHandlerImpl(
         return processActivity(claimsIdentity, conversationId, activityId, activity);
     }
 
+    /**
+     *
+     * @param claimsIdentity
+     * @param conversationId
+     * @param activityId
+     * @return deleteActivity()
+     */
```
Comment (Batta32) [inline libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java:109]: Can we add comments?

File: libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java [hunk 6]
```diff
@@ -101,6 +125,14 @@ public SkillHandlerImpl(
...
     }
 
+    /**
+     *
+     * @param claimsIdentity
+     * @param conversationId
+     * @param activityId
+     * @param activity
+     * @return ResourceResponse()
+     */
```
Comment (Batta32) [inline libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java:135]: Can we add comments?

File: libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java (hunk 3 above)
Comment (Batta32) [reply libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java:79]: Be careful with the `return processActivity()`, it's weird the comment

File: libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java (hunk 4 above)
Comment (Batta32) [reply libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java:94]: Be careful with the `return processActivity()`, it's weird the comment

File: libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java (hunk 5 above)
Comment (Batta32) [reply libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java:109]: Be careful with the `return deleteActivity()`, it's weird the comment

File: libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java (hunk 6 above)
Comment (Batta32) [reply libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java:135]: Be careful with the `return ResourceResponse()`, it's weird the comment

File: libraries/bot-connector/src/main/java/com/microsoft/bot/connector/authentication/ClaimsIdentity.java [hunk 7]
```diff
@@ -22,7 +22,7 @@ private ClaimsIdentity() {
 
     /**
      * Manually construct with auth issuer.
-     * 
+     *
```
Comment (Batta32) [inline libraries/bot-connector/src/main/java/com/microsoft/bot/connector/authentication/ClaimsIdentity.java:25]: Is this necessary?

File: libraries/bot-connector/src/main/java/com/microsoft/bot/connector/authentication/ClaimsIdentity.java [hunk 8]
```diff
@@ -31,7 +31,7 @@ public ClaimsIdentity(String withAuthIssuer) {
 
     /**
      * Manually construct with issuer and claims.
-     * 
+     *
```
Comment (Batta32) [inline libraries/bot-connector/src/main/java/com/microsoft/bot/connector/authentication/ClaimsIdentity.java:34]: Is this necessary?

File: libraries/bot-connector/src/main/java/com/microsoft/bot/connector/authentication/ClaimsIdentity.java [hunk 9]
```diff
@@ -54,7 +54,7 @@ public ClaimsIdentity(String withAuthIssuer, String withType, Map<String, String
 
     /**
      * Extract data from an auth0 JWT.
-     * 
+     *
```
Comment (Batta32) [inline libraries/bot-connector/src/main/java/com/microsoft/bot/connector/authentication/ClaimsIdentity.java:57]: Is this necessary?

File: libraries/bot-connector/src/main/java/com/microsoft/bot/connector/authentication/ClaimsIdentity.java [hunk 10]
```diff
@@ -68,7 +68,7 @@ public ClaimsIdentity(DecodedJWT jwt) {
 
     /**
      * Gets whether the claim is authenticated.
-     * 
+     *
```
Comment (Batta32) [inline libraries/bot-connector/src/main/java/com/microsoft/bot/connector/authentication/ClaimsIdentity.java:71]: Is this necessary?

File: libraries/bot-connector/src/main/java/com/microsoft/bot/connector/authentication/ClaimsIdentity.java [hunk 11]
```diff
@@ -77,7 +77,7 @@ public boolean isAuthenticated() {
 
     /**
      * The claims for this identity.
-     * 
+     *
```
Comment (Batta32) [inline libraries/bot-connector/src/main/java/com/microsoft/bot/connector/authentication/ClaimsIdentity.java:80]: Is this necessary?

File: libraries/bot-connector/src/main/java/com/microsoft/bot/connector/authentication/ClaimsIdentity.java [hunk 12]
```diff
@@ -86,7 +86,7 @@ public boolean isAuthenticated() {
 
     /**
      * The issuer.
-     * 
+     *
```
Comment (Batta32) [inline libraries/bot-connector/src/main/java/com/microsoft/bot/connector/authentication/ClaimsIdentity.java:89]: Is this necessary?

File: libraries/bot-connector/src/main/java/com/microsoft/bot/connector/authentication/ParameterizedBotFrameworkAuthentication.java [hunk 13]
```diff
@@ -222,24 +232,27 @@ public BotFrameworkClient createBotFrameworkClient() {
...
 
     // The following code is based on SkillValidation.AuthenticateChannelToken
-    private CompletableFuture<ClaimsIdentity> skillValidation_AuthenticateChannelToken(String authHeader, String channelId) {
+    private CompletableFuture<ClaimsIdentity> skillValidationAuthenticateChannelToken(String authHeader,
+                                                                                      String channelId) {
         TokenValidationParameters tokenValidationParameters = new TokenValidationParameters();
         tokenValidationParameters.validateIssuer = true;
         tokenValidationParameters.validIssuers = Arrays.asList(
-            // TODO: presumably this table should also come from configuration
+            // TO-DO: presumably this table should also come from configuration.
```
Comment (Batta32) [inline libraries/bot-connector/src/main/java/com/microsoft/bot/connector/authentication/ParameterizedBotFrameworkAuthentication.java:255]: Try with `//TODO`

File: libraries/bot-connector/src/main/java/com/microsoft/bot/connector/authentication/ParameterizedBotFrameworkAuthentication.java [hunk 14]
```diff
@@ -249,21 +262,21 @@ public BotFrameworkClient createBotFrameworkClient() {
         );
         tokenValidationParameters.validateAudience = false; // Audience validation takes place manually in code.
         tokenValidationParameters.validateLifetime = true;
-        tokenValidationParameters.clockSkew = Duration.ofMinutes(5);
+        tokenValidationParameters.clockSkew = Duration.ofMinutes(FIVE);
         tokenValidationParameters.requireSignedTokens = true;
 
-        // TODO: what should the openIdMetadataUrl be here?
+        // TO-DO: what should the openIdMetadataUrl be here?
```
Comment (Batta32) [inline libraries/bot-connector/src/main/java/com/microsoft/bot/connector/authentication/ParameterizedBotFrameworkAuthentication.java:268]: Try with `//TODO`

File: libraries/bot-connector/src/main/java/com/microsoft/bot/connector/authentication/ParameterizedBotFrameworkAuthentication.java [hunk 15]
```diff
@@ -304,29 +321,32 @@ public BotFrameworkClient createBotFrameworkClient() {
...
         toBotFromEmulatorTokenValidationParameters.validateIssuer = true;
         toBotFromEmulatorTokenValidationParameters.validIssuers = Arrays.asList(
-            // TODO: presumably this table should also come from configuration
-            "https://sts.windows.net/d6d49420-f39b-4df7-a1dc-d59a935871db/",                    // Auth v3.1, 1.0 token
-            "https://login.microsoftonline.com/d6d49420-f39b-4df7-a1dc-d59a935871db/v2.0",      // Auth v3.1, 2.0 token
-            "https://sts.windows.net/f8cdef31-a31e-4b4a-93e4-5f571e91255a/",                    // Auth v3.2, 1.0 token
-            "https://login.microsoftonline.com/f8cdef31-a31e-4b4a-93e4-5f571e91255a/v2.0",      // Auth v3.2, 2.0 token
-            "https://sts.windows.net/cab8a31a-1906-4287-a0d8-4eef66b95f6e/",                    // Auth for US Gov, 1.0 token
-            "https://login.microsoftonline.us/cab8a31a-1906-4287-a0d8-4eef66b95f6e/v2.0"        // Auth for US Gov, 2.0 token
+            // TO-DO: presumably this table should also come from configuration
```
Comment (Batta32) [inline libraries/bot-connector/src/main/java/com/microsoft/bot/connector/authentication/ParameterizedBotFrameworkAuthentication.java:329]: Try with `//TODO`

File: libraries/bot-connector/src/main/java/com/microsoft/bot/connector/authentication/PasswordServiceClientCredentialFactory.java [hunk 16]
```diff
@@ -106,7 +106,7 @@ public PasswordServiceClientCredentialFactory(String withAppId, String withPassw
 
         if (loginEndpoint.toLowerCase()
                 .startsWith(AuthenticationConstants.TO_CHANNEL_FROM_BOT_LOGIN_URL_TEMPLATE.toLowerCase())) {
-            // TODO: Unpack necessity of these empty credentials based on the loginEndpoint
+            // TO-DO: Unpack necessity of these empty credentials based on the loginEndpoint
```
Comment (Batta32) [inline libraries/bot-connector/src/main/java/com/microsoft/bot/connector/authentication/PasswordServiceClientCredentialFactory.java:109]: Try with `//TODO`

**Review by Batta32 — CHANGES_REQUESTED** [review]

File: libraries/bot-builder/src/main/java/com/microsoft/bot/builder/ChannelServiceHandler.java [hunk 17]
```diff
@@ -58,6 +57,8 @@ public ChannelServiceHandler(
      * HttpClient)} , we should move this code somewhere in that library when
      * we refactor auth, for now we keep it private to avoid adding more public
      * static functions that we will need to deprecate later.
+     * @param authHeader The Bearer token included as part of the request
```
Comment (Batta32) [inline libraries/bot-builder/src/main/java/com/microsoft/bot/builder/ChannelServiceHandler.java:60]:
```suggestion
     * @param authHeader The Bearer token included as part of the request.
```

File: libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java [hunk 18]
```diff
@@ -68,13 +70,25 @@ public SkillHandlerImpl(
...
     public CompletableFuture<ResourceResponse> onSendToConversation(
         ClaimsIdentity claimsIdentity,
         String conversationId,
         Activity activity) {
         return processActivity(claimsIdentity, conversationId, null, activity);
     }
 
+    /**
+     * {@inheritDoc}
+     */
```
Comment (Batta32) [inline libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java:91]: This class is not extending from another.. so the `inheritDoc` it's not okay. We should populate the commentaries.

File: libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java [hunk 19]
```diff
@@ -83,6 +97,9 @@ public SkillHandlerImpl(
         return processActivity(claimsIdentity, conversationId, activityId, activity);
     }
 
+    /**
+     * {@inheritDoc}
+     */
```
Comment (Batta32) [inline libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java:102]: This class is not extending from another.. so the `inheritDoc` it's not okay. We should populate the commentaries.

File: libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java [hunk 20]
```diff
@@ -101,6 +118,9 @@ public SkillHandlerImpl(
             callback);
     }
 
+    /**
+     * {@inheritDoc}
+     */
```
Comment (Batta32) [inline libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java:123]: This class is not extending from another.. so the `inheritDoc` it's not okay. We should populate the commentaries.

**Review by Batta32 — CHANGES_REQUESTED** [review]

File: libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java (hunk 18 above)
Comment (matiasroldan6) [reply libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java:91]: The "Impl" part really made me fall for it

Comment (matiasroldan6) [reply libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java:91]: sometime later we should fix [this one](https://github.com/southworks/botbuilder-java/blob/internal/feature/southworks/cloudadapter/base/libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/CloudSkillHandler.java#L140) too

Comment (Batta32) [reply libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java]: But that one is extending from the `CloudChannelServiceHandler`

File: libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java [hunk 21]
```diff
@@ -83,6 +103,14 @@ public SkillHandlerImpl(
         return processActivity(claimsIdentity, conversationId, activityId, activity);
     }
 
+    /**
+     * @param claimsIdentity  claimsIdentity for the bot, should have
+     *                        AudienceClaim, AppIdClaim and ServiceUrlClaim.
+     * @param conversationId  Conversation ID.
+     * @param activityId      activityId to delete.
+     *
+     * @return Task for a resource response.
```
Comment (Batta32) [inline libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java:112]: This is void so it should not return a `resource response`

**Review by Batta32 — CHANGES_REQUESTED** [review]

File: libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java (hunk 18 above)
Comment (matiasroldan6) [reply libraries/bot-builder/src/main/java/com/microsoft/bot/builder/skills/SkillHandlerImpl.java]: yes but the method is not there, or in the parent, or anywhere. Well, I didn't find it

**Review by Batta32 — APPROVED** [review]

**Pull request merged**